sygnal/sygnal.py

```diff
@@ -210,9 +210,25 @@ def run(self):
         bind_addresses = self.config["http"]["bind_addresses"]
         pushgateway_api = PushGatewayApiServer(self)
 
-        ensureDeferred(
-            self._make_pushkins_then_start(port, bind_addresses, pushgateway_api)
-        )
+        @defer.inlineCallbacks
+        def start():
+            try:
+                yield ensureDeferred(
+                    self._make_pushkins_then_start(
+                        port, bind_addresses, pushgateway_api
+                    )
+                )
+            except Exception:
+                # Print the exception and bail out.
+                print("Error during startup:", file=sys.stderr)
```

Comment:
Shouldn't this be `logger.error` or something similar? Or maybe we're assuming that logging doesn't work because of https://github.com/matrix-org/sygnal/issues/120, so we're only printing for now?

Reply:
You cannot guarantee that logging has been configured yet. At least I assume that's why the Synapse code does this.

Reply:
Why? Logging is configured in the `Sygnal` constructor, so it has to have been configured when calling `sygnal.run`.

Reply:
It is a pattern I've seen before, I think it guards against:
* Logging not being configured
* Logging being configured in a broken way (e.g. #120)

If I use `logger.error` here I see nothing in the logs, but this is likely #120 as you pointed out.

Reply:
sgtm